src/components/configMaps/ConfigMap.tsx

```diff
@@ -233,35 +235,75 @@ export function ConfigMapForm({ id, appId, name = "", external, data = null, typ
     async function handleDelete() {
         try {
             if (!envId) {
-                const { result } = await deleteConfig(id, appId, name)
+                await deleteConfig(id, appId, name)
                 toast.success('Successfully deleted')
                 updateForm(listIndex, null)
             }
             else {
-                const { result } = await deleteEnvironmentConfig(id, appId, envId, name);
+                await deleteEnvironmentConfig(id, appId, envId, name);
                 toast.success('Successfully deleted')
                 updateForm(true)
             }
-
         }
         catch (err) {
             showError(err)
         }
     }
 
+    function handleFilePermission(e): void {
+        let permissionValue = e.target.value;
+        setFilePermissionValue({ value: permissionValue, error: "" });
+    }
+
     async function handleSubmit(e) {
+        const configmapNameRegex = new RegExp(PATTERNS.CONFIGMAP_AND_SECRET_NAME);
         if (!configName.value) {
-            setName({ value: "", error: 'Field is manadatory' })
+            setName({ value: "", error: 'This is a required field' })
             return
         }
-        if (!/^[-.a-zA-Z0-9]+$/.test(configName.value)) {
-            setName({ value: configName.value, error: 'Name must be of format /^[-.a-zA-Z0-9]+$/' })
+        if (configName.value.length > 253) {
+            setName({ value: configName.value, error: 'More than 253 characters are not allowed' })
```

Comment:
check with Vikram, this is correct if we dont append anything to name, otherwise need to reduce this size

Reply:
No, he is not appending anything.